samanthas_telegram_bot/data_structures/context_types.py

```diff
@@ -0,0 +1,176 @@
+    """Class for data needed for every conversation and loaded once at application start."""
+
+    def __init__(self) -> None:
+        # can't be imported at top of the module: it will lead to circular import error
+        from samanthas_telegram_bot.api_queries.bot_init import (
+            get_age_ranges,
+            get_assessments,
+            get_day_and_time_slots,
+            get_languages_and_levels,
+        )
+
+        self.age_ranges_for_type = get_age_ranges()
+        self.assessment_for_age_range_id = get_assessments(lang_code="en")
+
+        day_and_time_slots = get_day_and_time_slots()
+
+        self.day_and_time_slot_for_slot_id: dict[int, DayAndTimeSlot] = {
+            slot.id: slot for slot in day_and_time_slots
+        }
+        """Matches IDs of DayAndTimeSlot objects to DayAndTimeSLot objects themselves.
+        Needed for matching callback data to entire objects (e.g. to show the user later on
+        what they have chosen)
+         """
+
+        self.day_and_time_slots_for_day_index: dict[int, tuple[DayAndTimeSlot, ...]] = {
+            index: tuple(slot for slot in day_and_time_slots if slot.day_of_week_index == index)
+            for index in range(7)
+        }
+        """Matches indexes of days of the week to `DayAndTimeSlot` objects. We ask the user
+        time slots day by day, so for each day we have to select slots with the correct day index.
+         """
+
+        languages_and_levels = get_languages_and_levels()
+        unique_language_ids = {item.language_id for item in languages_and_levels}
+
+        # make sure English comes first as it has to be displayed first to the user
+        self.sorted_language_ids = ["en"] + sorted(unique_language_ids - {"en"})
+        """Language IDs sorted by language code (but English always comes first)."""
+
+        self.language_and_level_for_id: dict[int, LanguageAndLevel] = {
+            item.id: item for item in languages_and_levels
+        }
+        """Matches IDs of `LanguageAndLevel` objects to same `LanguageAndLevel` objects."""
+
+        # naming with "_objects" to avoid ambiguity of "languages_and_levels"
+        self.language_and_level_objects_for_language_id: dict[
+            str, tuple[LanguageAndLevel, ...]
+        ] = {
+            language_id: tuple(
+                language_and_level
+                for language_and_level in languages_and_levels
+                if language_and_level.language_id == language_id
+            )
+            for language_id in self.sorted_language_ids
+        }
+        """Matches IDs of teaching languages (strings like "en", "de" etc.) to tuples
+        of corresponding `LanguageAndLevel` objects."""
+
+        self.language_and_level_id_for_language_id_and_level: dict[tuple[str, str], int] = {
+            (item.language_id, item.level): item.id for item in languages_and_levels
+        }
+        """Matches a tuple of language_id and level to an ID of `LanguageAndLevel` object.
```

Comment:
Are you sure you need all these comments here? It looks like there are more comments than a code :D

Reply:
Well these are docstrings :) I want to be able to see them from anywhere in the code in my IDE :)